test/python_client_test.py

```diff
@@ -102,34 +103,62 @@ def test_success_get_srv_keyspace(self):
     with self.assertRaises(ValueError):
       small.keyspace_id_to_shard_name_for_db_type(0x6000000000000000, 'replica')
 
+  # A packed keyspace_id from the middle of the full keyrange.
+  KEYSPACE_ID_0X80 = struct.Struct('!Q').pack(0x80 << 56)
+
+  def _open_keyspace_ids_cursor(self):
+    return self.conn.cursor('keyspace', 'master',
+                            keyspace_ids=[self.KEYSPACE_ID_0X80])
+
+  def _open_keyranges_cursor(self):
+    kr = keyrange.KeyRange(keyrange_constants.NON_PARTIAL_KEYRANGE)
+    return self.conn.cursor('keyspace', 'master', keyranges=[kr])
+
+  def _open_batch_cursor(self):
+    return self.conn.cursor(
+        tablet_type='master', cursorclass=vtgate_cursor.BatchVTGateCursor)
+
```

Comment:
there is also StreamVTGateCursor. Since callerID needs to cover all cases, better to also test it here. The mocked server side API is ready there